examples/audio_server/server.cc

```diff
@@ -30,101 +31,113 @@ int main(int argc, char** argv) {
-        state.active_channels++;
-      });
-      threads[channel_id] = std::thread([&safe_state, channel_id]() {
-        std::cout << "Channel '" << channel_id << "' is online" << std::endl;
-        while(true) {
-          auto control = safe_state.WaitFor(
-              [channel_id](SharedState const& state) {
-                return state.channel_exists(channel_id);
-              },
-              [channel_id](SharedState & state) {
-                if (state.channel_control[channel_id] == cmd_kill){
-                  state.channel_control.erase(channel_id);
-                  state.active_channels--;
-                  return ControlSignal{"", true};
-                }
-                return ControlSignal{state.channel_control[channel_id], false};
-              },
-              // TODO: reduce this time (only for debug)
-              std::chrono::seconds(1));
-          if(control.stop) {
-            std::cout << "Channel '" << channel_id << "' has been stopped" << std::endl;
-            break;
-          }
 
-          if(!control.apply.empty()) {
-            std::cout << channel_id << "received signal " << control.apply << std::endl;
-            // TODO: handle control signals here
+        state.threads[channel_id] = std::thread([&safe_state, channel_id]() {
+          std::cout << "Channel '" << channel_id << "' is online" << std::endl;
+          while(true) {
+            auto control = safe_state.WaitFor(
+                [channel_id](SharedState const& state) {
+                  return state.channel_exists(channel_id);
+                },
+                [channel_id](SharedState & state) {
+                  auto iter = state.to_kill.find(channel_id);
+                  if (iter != state.to_kill.end()) {
+                    state.channel_control.erase(channel_id);
+                    state.to_kill.erase(iter);
+                    return ControlSignal{true};
+                  }
+                  return ControlSignal{false};
+                },
+                // TODO: reduce this time (only for debug)
+                std::chrono::seconds(1));
+            if(control.stop) {
+              std::cout << "Channel '" << channel_id << "' has been stopped" << std::endl;
+              break;
+            }
+
+            // TODO: route the stream here
+            // streaming_sockets example could perfectly fit there
+            std::cout << "[" << channel_id << "] worker tick" << std::endl;
+            std::this_thread::sleep_for(std::chrono::milliseconds(2000));
           }
-
-          // TODO: route the stream here
-          std::cout << "[" << channel_id << "] worker tick" << std::endl;
-
-          std::this_thread::sleep_for(std::chrono::milliseconds(2000));
-        }
+        });
+        return channel_validation{true, "created"};
       });
-      r("created\n");
+      r(node_state.msg, (node_state.is_valid ? HTTPResponseCode.OK : HTTPResponseCode.BadRequest));
```

Comment:
Yep, just `r(node_state [, ...])` would return the JSON as long as it's a `CURRENT_STRUCT`, ref. https://github.com/C5T/Current/blob/stable/blocks/http/test.cc#L74-L82

Reply:
Wow, that's cool!

Reply:
> Wow, that's cool!

Enjoy! There's also `class Response` FTR, so that if you want to `return` it from some lambda, and want to have different HTTP response codes and/or headers or content type, etc.